reference/best-practices/java-app.md

```diff
@@ -0,0 +1,362 @@
+- You can use the query interface with `ResultHandler` to avoid getting the entire result set at once.
+- You can use the `Cursor` class for stream reading.
+
+If you configure mappings using XML, you can stream read results by configuring `fetchSize="-2147483648"`(`Integer.MIN_VALUE`) in the mapping's `<select>` section.
+
+```xml
+<select id="getAll" resultMap="postResultMap" fetchSize="-2147483648">
+  select * from post;
+</select>
+```
+
+If you configure mappings using code, you can add the `@Options(fetchSize = Integer.MIN_VALUE)` annotation and keep the type of results as `Cursor` so that the SQL results can be read in streaming.
+
+```java
+@Select("select * from post")
+@Options(fetchSize = Integer.MIN_VALUE)
+Cursor<Post> queryAllPost();
+```
+
+### `ExecutorType`
+
+You can choose `ExecutorType` during `openSession`. MyBatis supports three types of executors:
+
+- Simple: The Prepared Statements are called to JDBC for each execution (if the JDBC configuration item `cachePrepStmts` is enabled, repeated Prepared Statements will be reused)
+- Reuse: The Prepared Statements are cached in `executor`, so that you can reduce duplicate calls for Prepared Statements without using the JDBC `cachePrepStmts`
+- Batch: Each update operation (`INSERT`/`DELETE`/`UPDATE`) will first be added to the batch, and will be executed until the transaction commits or a `SELECT` query is performed. If `rewriteBatchStatements` is enabled in the JDBC layer, it will try to rewrite the statements. If not, the statements will be sent one by one.
+
+Usually, the default value of `ExecutorType` is `Simple`. You need to change `ExecutorType` when calling `openSession`. If it is batch execution, you might find that in a transaction the update or insert statements are executed pretty fast, but it is slower when reading data or committing the transaction. This is actually normal, so you need to note this when troubleshooting slow SQL queries.
+
+## Spring Transaction
+
+In the real world, applications might use [Spring Transaction](https://docs.spring.io/spring/docs/4.2.x/spring-framework-reference/html/transaction.html) and AOP aspects to start and stop transactions.
+
+By adding the `@Transactional` annotation to the method definition, AOP starts the transaction before the method is called, and commit the transaction before the method returns the result. If your application has a similar need, you can find `@Transactional` in code to determine when the transaction is started and closed.
+
+Pay attention to a special case of embedding. If it occurs, Spring will behave differently based on the [Propagation](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/Propagation.html) configuration. Because TiDB does not support savepoint, nested transactions are not supported yet.
+
+## Misc
+
```

Comment:
Please add one or two sentences to briefly introduce the following section.

Reply:
Added